larray_editor/utils.py

```diff
@@ -595,3 +595,51 @@ def get_sample(data, maxsize):
 def get_sample_indices(data, maxsize):
     flat_indices = np.arange(0, data.size, get_sample_step(data, maxsize))
     return np.unravel_index(flat_indices, data.shape)
+
+
+class RecentFileList(object):
+    MAX_RECENT_FILES = 10
+
+    def __init__(self, list_name, actions=False, parent=None):
+        self.settings = QSettings()
+        self.list_name = list_name
+        if self.settings.value(list_name) is None:
+            self.settings.setValue(list_name, [])
+        self.actions = [QAction(parent) for _ in range(self.MAX_RECENT_FILES)] if actions else None
+
+    @property
+    def recent_files(self):
+        return self.settings.value(self.list_name)
+
+    @recent_files.setter
+    def recent_files(self, files):
+        self.settings.setValue(self.list_name, files[:self.MAX_RECENT_FILES])
+        self.update_actions()
+
+    def add(self, filepath):
+        if filepath is not None:
+            recent_files = self.recent_files
+            if filepath in recent_files:
+                recent_files.remove(filepath)
+            recent_files = [filepath] + recent_files
+            self.recent_files = recent_files
+            self.update_actions()
+
+    def clear(self):
+        self.recent_files = []
+
+    def update_actions(self):
```

Comment:
make this a private method?

Reply:
This method is call in `MappingEditor.setup_menu_bar`

Reply:
I know, but this should not be needed IMO (if actions was a either property, or updated automatically whenever the filelist changes).